modules/imgcodecs/src/loadsave.cpp

```diff
@@ -279,7 +279,7 @@ imread_( const String& filename, int flags, int hdrtype, Mat* mat=0, int scale_d
 
     // grab the decoded type
     int type = decoder->type();
-    if( flags != IMREAD_UNCHANGED )
+    if( (flags & IMREAD_LOAD_GDAL) == 0 && flags != IMREAD_UNCHANGED )
```

Comment:
I guess the problem it's here. Added this hack since for the image that I'm using ( 145x145x200 ) it works.